azurerm/internal/services/apimanagement/api_management_api_resource.go

```diff
@@ -175,6 +175,53 @@ func resourceArmApiManagementApi() *schema.Resource {
 				Default:  false,
 			},
 
+			"oauth2_authorization": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"authorization_server_name": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validate.ApiManagementChildName,
+						},
+						"scope": {
+							Type:     schema.TypeString,
+							Optional: true,
+							// There is currently no validation, as any length and characters can be used in the field
```

Comment:
Is this allowed to be an empty string?

Reply:
Yes, and even a multi-kilobyte string, or null, or anything apparently.

Reply:
Since there are no docs on the valid values for the field, and practical trial has shown that null or a string of any length is valid, I don't have any better ideas.

Reply:
> Since there are no docs on the valid values for the field, and practical trial has shown that null or a string of any length is valid, I don't have any better ideas.

That's as exhaustive as I think you can reasonably be, thanks for the diligence!